common/cluster/metadata.go

```diff
@@ -219,12 +215,14 @@ func (metadata *metadataImpl) GetAllClientAddress() map[string]config.Address {
 	return metadata.clusterToAddress
 }
 
-// IsArchivalEnabled whether archival is enabled
-func (metadata *metadataImpl) IsArchivalEnabled() bool {
-	return metadata.enableArchival()
-}
-
-// GetDefaultArchivalBucket returns the default archival bucket name
-func (metadata *metadataImpl) GetDefaultArchivalBucket() string {
-	return metadata.defaultArchivalBucket
+// ArchivalConfig returns the archival config of the cluster.
+// The cluster is considered disabled for archival if either there is no default bucket or if config indicates disabled.
+// If status is paused or enabled then default bucket is set.
+func (metadata *metadataImpl) ArchivalConfig() (retCfg *ArchivalConfig) {
+	bucketSet := len(metadata.defaultBucket) != 0
+	status := GetArchivalStatus(metadata.archivalStatus())
+	if !bucketSet || status == ArchivalDisabled {
```

Comment:
I believe this validation already lives inside your ArchivalConfig class - so this code is redundant. And I like this code living inside the ArchivalConfig class